WooCommerce/Classes/ViewRelated/InAppPurchases/InAppPurchasesForWPComPlansManager.swift

```diff
@@ -73,8 +73,8 @@ final class InAppPurchasesForWPComPlansManager: InAppPurchasesForWPComPlansProto
         }
     }
 
-    func purchaseProduct(with id: String, for remoteSiteId: Int64) async throws {
-        _ = try await withCheckedThrowingContinuation { continuation in
+    func purchaseProduct(with id: String, for remoteSiteId: Int64) async throws -> StoreKit.Product.PurchaseResult {
```

Comment:
It would be great to keep `StoreKit` references inside the boundaries of the IAP interface if possible, so their clients don't have to even import it. That way we have `WPComPlanProduct` that acts as a facade of the StoreKit.Product, hiding any details about `StoreKit`. Can we do the same with `StoreKit.Product.PurchaseResult`? With a protocol or a typealias? 

`typealias PurchaseResult = StoreKit.Product.PurchaseResult`

Reply:
Oh good call, I'll add that

Reply:
Added in [ded8c4c](https://github.com/woocommerce/woocommerce-ios/pull/8133/commits/ded8c4c9513f2fe988198d436988b337459827ed)